### ISSUE TITLE
R-scripts nuked by osx-arm migrator

### ISSUE THREAD

**Issue by h-vetinari** [regro/cf-scripts#2927]
The bot added the following diff in https://github.com/conda-forge/r-arrow-feedstock/pull/88:
```diff
--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -1,16 +1,6 @@
 #!/bin/bash
-set -ex

 export DISABLE_AUTOBREW=1

-# arrow uses C++17
-export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -std=c++17"
-export LIBARROW_BUILD=false
-
-if [[ "${target_platform}" == osx-* ]]; then
-    # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
-    export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
-fi
-
-# ${R_ARGS} necessary to support cross-compilation
-${R} CMD INSTALL --build r/. ${R_ARGS}
+# shellcheck disable=SC2086
+${R} CMD INSTALL --build . ${R_ARGS}
```

which looks like it's blindly (and incorrectly) overwriting the build script - @beckermr asked me to open an issue here.

I've looked at https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/arch.py, but cannot figure out why the osx-arm migrator would do that. Also of note, the linux-arches [PR](https://github.com/conda-forge/r-arrow-feedstock/pull/91) for the same branch did not cause such changes to the build script.

**Comment by beckermr** [comment]
I expect this may be a minimigrator and not the primary one.

**Comment by h-vetinari** [comment]
> I expect this may be a minimigrator and not the primary one.

Yeah, but I couldn't find which one would do that. At least, none of the names in https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators stick out.

**Comment by beckermr** [comment]
This happens here: https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/cross_compile.py#L365.

cc @isuruf for advice on how to proceed.

**Comment by beckermr** [comment]
Likely related to the comment @isuruf made here: https://github.com/regro/cf-scripts/pull/2946#issuecomment-2305321959
